task5/func2.js

```diff
@@ -1,15 +1,15 @@
-class ReversePromise {
+class ReversePromise extends Promise {
     constructor(callback) {
+        super(callback);
         this.arrayCbs = [];
-        this.promise = new Promise(callback).finally(() => {
-            this.arrayCbs.forEach(element => {
-                element();
-            });
-        });
+        setTimeout(() => {
+            this.arrayCbs.reverse().forEach(callback => callback());
+            this.arrayCbs = [];
+        }, 0);
```

Comment:
It's better to move to 'then' method because the constructor basically should have only definitions of properties :)